src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -940,7 +942,7 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             // so we need to ensure that arguments cannot use stack temps
             BoundExpression right = node.Right;
             bool mayPushReceiver = (right.Kind == BoundKind.ObjectCreationExpression &&
-                right.Type.IsVerifierValue() &&
+                ((BoundObjectCreationExpression)right).Type.IsVerifierValue() &&
```

Comment:
>(BoundObjectCreationExpression) [](start = 17, length = 31)

No need to introduce this cast

Reply:
The case results in calling a different `Type` property.